File: Lab 2/Danial Asghar/Kafka-Node/topic.js
const {Kafka} = require("kafkajs")
run();

async function run(){
    try {
        
        const kafka = new Kafka({
            "clientId": "myapp",
            "brokers": ["localhost:9093", "localhost:9094", "localhost:9095"]
        })

        const admin = kafka.admin();
        console.log("Connecting....")
        await admin.connect();
        console.log("Connected.")
        
        await admin.createTopics({
            "topics": [{
                "topic" : "Users",
                "numPartitions": 2
            }]
        })
        console.log("Topics Created!")

        admin.disconnect()

    } catch (error) {
        console.log(`Something bad happened ${error}`)
    }
    finally{
        process.exit(0)
    }
}

File: Lab 2/Owais Quadri/Python/topic.py
from kafka.admin import KafkaAdminClient, NewTopic
topic="Usernames"
admin=KafkaAdminClient(
    bootstrap_servers=["localhost:9093","localhost:9094","localhost:9095"],
    client_id='myapp'
)
print("Creating Topic ",topic)
topic_list = [NewTopic(name=topic, num_partitions=2, replication_factor=2)]
admin.create_topics(new_topics=topic_list, validate_only=False)
print("Topic '",topic,"' Created.")